### DISCUSSION TITLE
shorthand `-> exp` syntax

### DISCUSSION THREAD

**Discussion by JonathanStarup** [flix/flix#8472] [Ideas]
In order for thunking to be prettier, I vote to introduce the `-> exp` syntax. This could be shorthand for either `() -> exp` or `_ -> exp`

Examples:
```scala
myWhile(() -> condition)(() -> { body })
// vs
myWhile(-> condition)(-> { body })
```
```scala
opt |> getWithDefault(() -> default)
// vs
opt |> getWithDefault(-> default)
```

One complication is with stmt thunks like
```
while(-> true)(-> {let x = 42; x})
```
where double parens are (naively) needed. But we can add `(-> stmt)` as its own parsing rule, allowing
```
while(-> true)(-> let x = 42; x)
```
to parse correctly

**Comment by magnus-madsen** [comment]
I agree its nicer, but is it nice enough?

**Reply by JonathanStarup** [reply]
I've added a new idea to it, now I think it is. I think this version is as nice as it gets, withing removing the explicit arrow (which is think is a very good thing to keep)
